pyrobosim/pyrobosim/core/room.py

```diff
@@ -71,6 +74,29 @@ def __init__(
         else:
             self.nav_poses = [Pose.from_list(self.centroid)]
 
+    def _parse_color(self, color):
+        """
+        Parses a color input and returns an RGB tuple.
+
+        :param color: Input color as a list, string, or hexadecimal.
+        :type color: list[float] | str
+        :return: RGB tuple in range (0.0, 1.0).
+        :rtype: list[float]
+        """
+        if isinstance(color, list) and len(color) == 3:
+            return color
+        elif isinstance(color, str):
+            if color in CSS4_COLORS:
+                return list(to_rgb(CSS4_COLORS[color]))
+            try:
```

Comment:
Is therr a way to avoid using a try/catch by checking things? I try not to use them in my code unless absolutely necessary

Reply:
Implemented using regex to avoid try/catch.